tests/jballerina-unit-test/src/test/resources/test-src/statements/compoundassignment/compound_assignment.bal

```diff
@@ -1200,6 +1200,12 @@ function testCompoundAssignmentBitwiseXOROperation() {
     assertEqual(a11, 9);
 }
 
+function testCompoundAssignmentMemberAccess() {
+       map<int>? m = {x:2};
+       m["x"] += 1;
```

Comment:
We need to disallow this at compile time, right? Looping in @hasithaa also.

Reply:
yes, This needs to be a compile-time error.

Reply:
Is it finalized? and are we adding that via this PR or separate PR?

Reply:
See last comment from James, https://github.com/ballerina-platform/ballerina-spec/issues/191

Reply:
Let's add this to the same PR.

Reply:
Added